syft/execution/placeholder.py

```diff
@@ -21,11 +29,72 @@ def __init__(self, owner=None, id=None, tags: set = None, description: str = Non
         """
         super().__init__(id=id, owner=owner, tags=tags, description=description)
 
-        if not isinstance(self.id, PlaceholderId):
-            self.id = PlaceholderId(self.id)
+        if not isinstance(self.id, syft.execution.placeholder_id.PlaceholderId):
+            self.id = syft.execution.placeholder_id.PlaceholderId(self.id)
 
         self.expected_shape = tuple(shape) if shape is not None else None
         self.child = None
+        self.role = role
+        self.tracing = tracing
+
+    def get_class_attributes(self):
+        """
+        Specify all the attributes need to build a wrapper correctly when returning a response.
+        """
+        return {"role": self.role, "tracing": self.tracing, "owner": self.owner}
+
+    @classmethod
+    def handle_func_command(cls, command):
+        """ Receive an instruction for a function to be applied on a Placeholder,
+        Replace in the args with their child attribute, forward the command
+        instruction to the handle_function_command of the type of the child attributes,
+        get the response and wrap it in a Placeholder.
+        We use this method to perform the tracing.
+
+        Args:
+            command: instruction of a function command: (command name,
+                <no self>, arguments[, kwargs])
+
+        Returns:
+            the response of the function command
+        """
+        cmd, _, args, kwargs = command
+
+        # Replace all PlaceHolders with their child attribute
+        new_args, new_kwargs, new_type = hook_args.unwrap_args_from_function(cmd, args, kwargs)
+
+        # build the new command
+        new_command = (cmd, None, new_args, new_kwargs)
+
+        # Send it to the appropriate class and get the response
+        response = new_type.handle_func_command(new_command)
+
+        # Find first placeholder in args
+        ph_arg = None
+        for arg in args:
+            # TODO throw an error or handle the case where no placeholder provided in args
+            if isinstance(arg, PlaceHolder):
+                ph_arg = arg
+
```

Comment:
I suppose this could implemented with a check after the loop:
```suggestion
        # Find first placeholder in args
        ph_arg = None
        for arg in args:
            if isinstance(arg, PlaceHolder):
                ph_arg = arg

        if ph_arg is None:
            raise SomeException
```

Reply:
Hum, after reflexion, is it possible that no PlaceHolder is there? Being in PlaceHolder.handle_func_command, maybe we don't need any check

Reply:
Makes sense to me. Let's leave it off.